app/src/main/java/ingsw/group1/findmyphone/cryptography/SMSCipher.java

```diff
@@ -0,0 +1,135 @@
+package ingsw.group1.findmyphone.cryptography;
+
+import android.content.Context;
+
+import ingsw.group1.msglibrary.SMSMessage;
+import ingsw.group1.msglibrary.SMSPeer;
+
+/**
+ * Class used to cipher and decipher a SMSMessage.
+ *
+ * @author Pardeep Kumar
+ */
+public class SMSCipher implements MessageCipher<SMSMessage> {
+
+    private static final char PARSING_CHARACTER = '0';
+    private PasswordManager passwordManager;
+
+
+    public SMSCipher(Context context) {
```

Comment:
I just realized, how do we encrypt and decrypt with other users' passwords, which are not stored anywhere?
Sorry for not realizing this earlier but it might be better to have SMSCiper as just a "parser" with the password as its parameters, otherwise, we can only encrypt and decrypt with our stored password.

Reply:
Yes that's actually the best solution .